community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java

```diff
@@ -50,20 +53,29 @@ public SchemaKernelException( Status statusCode, String message )
         super( statusCode, message );
     }
 
-    protected static String messageWithLabelAndPropertyName( TokenNameLookup tokenNameLookup, String formatString,
+    static String messageWithLabelAndPropertyName( TokenNameLookup tokenNameLookup, String formatString,
             LabelSchemaDescriptor descriptor )
     {
+        int[] propertyIds = descriptor.getPropertyIds();
+
         if ( tokenNameLookup != null )
         {
+            String propertyString = propertyIds.length == 1 ?
+                                    "property " + tokenNameLookup.propertyKeyGetName( descriptor.getPropertyId() ) :
+                                    "properties " + Arrays.stream( propertyIds )
+                                            .mapToObj( (i) -> "'" + tokenNameLookup.propertyKeyGetName( i ) + "'")
+                                            .collect( Collectors.joining( " and " ));
             return String.format( formatString,
-                    tokenNameLookup.labelGetName( descriptor.getLabelId() ),
-                    tokenNameLookup.propertyKeyGetName( descriptor.getPropertyId() ) );
+                    tokenNameLookup.labelGetName( descriptor.getLabelId() ), propertyString);
         }
         else
         {
+            String keyString = propertyIds.length == 1 ? "key[" + propertyIds[0] + "]" :
+                               "keys[" + Arrays.stream( propertyIds )
```

Comment:
We also seem to be solving this in multiple places. See for example https://github.com/neo4j/neo4j/blob/f8512e67e1533d675e2f045fdb027391ed4b731b/community/kernel/src/main/java/org/neo4j/kernel/impl/util/dbstructure/DbStructureArgumentFormatter.java#L147